bundles/org.openhab.binding.heliosventiremote/src/main/java/org/openhab/binding/heliosventiremote/internal/HeliosVentiRemoteHandler.java

```diff
@@ -0,0 +1,510 @@
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.io.transport.serial.PortInUseException;
+import org.eclipse.smarthome.io.transport.serial.SerialPort;
+import org.eclipse.smarthome.io.transport.serial.SerialPortEvent;
+import org.eclipse.smarthome.io.transport.serial.SerialPortEventListener;
+import org.eclipse.smarthome.io.transport.serial.SerialPortIdentifier;
+import org.eclipse.smarthome.io.transport.serial.SerialPortManager;
+import org.eclipse.smarthome.io.transport.serial.UnsupportedCommOperationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HeliosVentiRemoteHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Raphael Mack - Initial contribution
+ */
+@NonNullByDefault
+public class HeliosVentiRemoteHandler extends BaseThingHandler implements SerialPortEventListener {
+    private static final int BUSMEMBER_MAINBOARD = 0x11;
+    private static final int BUSMEMBER_SLAVEBOARDS = 0x10;
+    private static final byte BUSMEMBER_CONTROLBOARDS = (byte) 0x20;
+    private static final int BUSMEMBER_REC_MASK = 0xF0; // interpreting frames delivered to BUSMEMBER_ME &
+                                                        // BUSMEMBER_REC_MASK
+    private static final int BUSMEMBER_ME = 0x2F; // used as sender when communicating with the helios system
+
+    private static final HashMap<Byte, HeliosVentiRemoteVariable> variables = new HashMap();
+    private static final int POLL_OFFLINE_THRESHOLD = 3;
+
+    private final Logger logger = LoggerFactory.getLogger(HeliosVentiRemoteHandler.class);
+
+    private @Nullable HeliosVentiRemoteConfiguration config;
+
+    private @NonNullByDefault({}) SerialPortManager serialPortManager;
+
+    private String serialPortName;
+    private @Nullable SerialPortIdentifier portId;
+    private @Nullable SerialPort serialPort;
+    private @Nullable InputStream inputStream;
+    private @Nullable OutputStream outputStream;
+
+    // Polling variables
+    public int pollPeriod = 0;
+    private @Nullable ScheduledFuture<?> pollingTask;
+    private int pollCounter;
+
+    public HeliosVentiRemoteHandler(Thing thing, final SerialPortManager serialPortManager) {
+        super(thing);
+        this.serialPortManager = serialPortManager;
+        serialPortName = "";
+
+        // Read-only
+        variables.put((byte) 0x32,
```

Comment:
I do not fully get your point and how the bridge stuff is linked to the quoted lines above. Maybe the name is just misleading? The "variables" are just the definition of the datapoints, that are readable, some are writable and they have some type to define their interpretation.

In case we would allow multiple things on the same RS485 bus having a bridge would definitly be the way to go. In this case on the bus there are only one master device - the KWL - and 1 or more remote controls. Supporting the remotes as separate things doesn't seem interesting in my eyes, so we basically have exactly one thing to talk to on the bus. Supporting multiple such buses with different RS485 interfaces seems a rarely interesting case but it should even work with the current code I guess.
So is a bridge really worth the effort (also for the user to configure a bridge and a single thing on it)?

Reply:
Ok. I agree that you don't need the bridge in this case. I assumed there could be several devices on the bus. 

But if there is only one thing per port, why is ' variables' marked as' static'? You could make it a regular private field.

But in fact, I think you don't need it this way  at all. Aside from the ' channelUID' all parts are constants, the 'channelId' is also a constant. In 'handleCommand' you can get the 'channelId' from the 'channelUID'  and you can directly pass the ' channelId' to 'updateState'. The 'HashMap' could then be static and filled from a resourcefile. This is much cleaner than filling it on every thing construction.

Reply:
I probably just started to write "private static final" because it is a constant beast. My java experience is also somehow aged... If you have a pointer to an example on how to use a resource file for that I am more than happy to implement that solution.

Also a rename of variables to datapoints might be wise...

Reply:
That should be reworked now.